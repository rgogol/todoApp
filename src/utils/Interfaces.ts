export interface iReducerAction {
  type: string;
  payload: object | Array<object>;
}

export interface iReducerState {
  data: object | null;
  error: object | null;
  loading: boolean;
}

export interface iSagaOptions {
  payload: string | object;
  afterSagaSuccess?: void;
  afterSagaFailure?: void;
}

export interface iTask {
  id: number;
  value: string;
  isDone: boolean;
  date: string;
}
