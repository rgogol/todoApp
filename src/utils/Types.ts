import { iTask } from './Interfaces';

type ListTypes = {
  data: Array<iTask>,
};

export type TodosTypes = {
  list: ListTypes,
};
