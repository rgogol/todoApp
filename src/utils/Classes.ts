export class cTask {
  constructor(
    public id: number,
    public value: string,
    public isDone: boolean,
    public date: string,
  ) { }
}
