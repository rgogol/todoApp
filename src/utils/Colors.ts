export const WHITE = '#ffffff';
export const BLACK = '#000000';
export const LIGHT_BLUE = '#cfd6ea';
export const DARK_BLUE = '#4e598c';
export const SKY = '#102e4a';
export const LIGHT_GREY = '#f8f8f8';
export const TRANSPARENT = 'transparent';
export const BROWN = '#750d37';
