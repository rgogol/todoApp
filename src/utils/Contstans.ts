import { Dimensions } from 'react-native';

export const TODOS_KEY = 'todosKey';

export const WINDOW_WIDTH = Dimensions.get('window').width;
export const WINDOW_HEIGHT = Dimensions.get('window').height;
