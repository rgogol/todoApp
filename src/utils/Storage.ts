import AsyncStorage from '@react-native-community/async-storage';

export const readFromStore = async (keyName: string) => {
  try {
    const value: string | null = await AsyncStorage.getItem(keyName);
    const parseValue = value ? JSON.parse(value) : null;
    return parseValue;
  } catch (e) {
    return false;
  }
};

export const saveInStore = async (keyName: string, value: object) => {
  const encryptedString = JSON.stringify(value);

  await AsyncStorage.setItem(keyName, encryptedString);
};

export const removeFromStore = async (keyName: string) => {
  await AsyncStorage.removeItem(keyName);
};
