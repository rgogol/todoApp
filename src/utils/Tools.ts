export const getCurrentDate = () => {
  const date = new Date();
  const day = String(date.getDate()).padStart(2, '0');
  const month = String(date.getMonth() + 1).padStart(2, '0');
  const year = date.getFullYear();

  return `${day}/${month}/${year}`;
};

export const generateId = () => {
  const min: number = 0;
  const max: number = 99999;

  return Math.floor(Math.random() * (max - min) + min);
};
