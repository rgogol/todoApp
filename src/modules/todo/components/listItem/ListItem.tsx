import React from 'react';
import { View, Text, Image, TouchableOpacity } from 'react-native';
import { useDispatch, useSelector } from 'react-redux';
import { iTask } from '../../../../utils/Interfaces';
import { TodosTypes } from '../../../../utils/Types';
import { markAsDone } from '../../redux/actions';
import ListItemStyles from './styles';

const checkMarkSource = require('../../../../assets/check_mark.png');

const ListItem = ({ item }: { item: iTask }) => {
  const data = useSelector((state: TodosTypes) => state.list.data);
  const dispatch = useDispatch();

  const { value, date, isDone } = item;
  const {
    container,
    row,
    iconContainer,
    iconStyles,
    nameStyles,
    dateStyles,
    iconDoneStyles,
    containerDoneStyles,
  } = ListItemStyles;

  const onPressAction = () => {
    const task = { ...item, isDone: true };
    dispatch(markAsDone(data, task));
  };

  return (
    <View style={[container, row, isDone && containerDoneStyles]}>
      <View style={row}>
        <TouchableOpacity
          testID="mark-as-done-btn"
          style={[iconContainer, isDone && iconDoneStyles]}
          onPress={onPressAction}
          disabled={isDone}>
          {isDone && <Image source={checkMarkSource} style={iconStyles} />}
        </TouchableOpacity>
        <Text style={nameStyles}>{value}</Text>
      </View>
      <Text style={dateStyles}>{date}</Text>
    </View>
  );
};

export default ListItem;
