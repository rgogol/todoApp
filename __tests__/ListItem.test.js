import React from 'react';
import { shallow } from 'enzyme';
import ListItem from '../src/modules/todo/components/listItem';
import { cTask } from '../src/utils/Classes';
import { getCurrentDate } from '../src/utils/Tools';

const mockDispatch = jest.fn();
jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: () => mockDispatch,
}));

describe('ListItem', () => {
  const fakeData = new cTask(1, 'Drink water', false, getCurrentDate());
  const wrapper = shallow(<ListItem item={fakeData} />).dive();

  it('render as expected', () => {
    expect(wrapper).toMatchSnapshot();
  });

  // it('press mark as done', () => {
  //   const instance = wrapper.instance();
  //   const button = wrapper.find({ testID: 'mark-as-done-btn' });
  //   const onPress = jest.spyOn(button, 'onPressAction');

  //   instance.forceUpdate();
  //   button.props().onPress();
  //   expect(onPress).toHaveBeenCalledTimes(1);
  // });
});
