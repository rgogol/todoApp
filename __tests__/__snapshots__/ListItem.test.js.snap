// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem render as expected 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#f8f8f8",
        "borderRadius": 5,
        "elevation": 2,
        "justifyContent": "space-between",
        "marginHorizontal": 10,
        "marginTop": 10,
        "padding": 15,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 3,
      },
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      },
      false,
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <ForwardRef
      disabled={false}
      onPress={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderColor": "#4e598c",
            "borderRadius": 15,
            "borderWidth": 1,
            "height": 30,
            "justifyContent": "center",
            "marginRight": 10,
            "width": 30,
          },
          false,
        ]
      }
      testID="mark-as-done-btn"
    />
    <Text
      style={
        Object {
          "fontSize": 15,
          "fontWeight": "500",
        }
      }
    >
      Drink water
    </Text>
  </View>
  <Text
    style={
      Object {
        "color": "#4e598c",
        "fontSize": 13,
      }
    }
  >
    18/09/2020
  </Text>
</View>
`;
